pkg/volume/azure_file/azure_provision.go

```diff
@@ -138,7 +138,9 @@ func (a *azureFileProvisioner) Provision() (*v1.PersistentVolume, error) {
 
 	var sku, location, account string
 
-	name := volume.GenerateVolumeName(a.options.ClusterName, a.options.PVName, 75)
+	// File share name has a length limit of 63, and it cannot contain two consecutive '-'s.
+	name := volume.GenerateVolumeName(a.options.ClusterName, a.options.PVName, 63)
+	name = strings.Replace(name, "--", "-", -1)
 	capacity := a.options.PVC.Spec.Resources.Requests[v1.ResourceName(v1.ResourceStorage)]
```

Comment:
Some test/docs would be nice about why we are doing what we are doing and also making sure it doesn't regress.

Reply:
yes, I will add azure file test case in master branch first, since this is only a cherry pick, shall we merge this PR first?

Reply:
@gnufied agreed with andy, this is a cherry-pick aimed to fix a critical bug.

Reply:
The original PR was opened without cherry-pick title I think, so I didn't think it was a cherry-pick.